pydantic/main.py

```diff
@@ -207,6 +208,9 @@ def __new__(mcs, name, bases, namespace):
 _missing = object()
 
 
+T = TypeVar('T', bound='BaseModel')
```

Comment:
To make life easier for people in future, can you rename this `PydanticModel` (or just `Model`), also can you move the definition into `TYPE_CHECKING` above, then use the string `'PydanticModel'` below.